File: iosApp/iosApp/Presentation/Settings/SettingsUiState.swift
//
//  SettingsUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation

struct SettingsUiState {
    var isLoading: Bool = false
    var errorMessage: String?
    var appVersion: String = Self.resolvedAppVersion()
    var themeMode: AppThemeMode = .light
    var brandTheme: AppBrandTheme = .maidCafe

    static let empty = SettingsUiState()

    private static func resolvedAppVersion() -> String {
        guard let version = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String else {
            preconditionFailure("CFBundleShortVersionString is missing")
        }
        let trimmedVersion = version.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmedVersion.isEmpty else {
            preconditionFailure("CFBundleShortVersionString is empty")
        }
        return trimmedVersion
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/settings/SettingsUiState.kt
package com.hhp227.concafe.presentation.settings

import com.hhp227.concafe.presentation.theme.AppBannerLayout
import com.hhp227.concafe.presentation.theme.AppBrandTheme
import com.hhp227.concafe.presentation.theme.AppThemeMode

data class SettingsUiState(
    val isLoading: Boolean = false,
    val errorMessage: String? = null,
    val appVersion: String = currentAppVersion(),
    val themeMode: AppThemeMode = AppThemeMode.LIGHT,
    val brandTheme: AppBrandTheme = AppBrandTheme.MAID_CAFE,
    val bannerLayout: AppBannerLayout = AppBannerLayout.FULL_BLEED
) {
    companion object {
        fun empty(): SettingsUiState = SettingsUiState()
    }
}
